src/data_distribution/prime_mapp_distribution.hpp

```diff
@@ -0,0 +1,132 @@
+/**
+ * @file
+ * @brief This file implements the prime-mapp and prime-mapp-block data distributions
+ * @copyright Eta Scale AB. Licensed under the Eta Scale Open Source License. See the LICENSE file for details.
+ */
+
+#ifndef argo_prime_mapp_distribution_hpp
+#define argo_prime_mapp_distribution_hpp argo_prime_mapp_distribution_hpp
+
+namespace argo {
+	namespace data_distribution {
+		/**
+		 * @brief the prime-mapp data distribution
+		 * @details distributes pages using a two-phase round-robin strategy
+		 */
+		template<int instance>
+		class prime_mapp_distribution : public base_distribution<instance> {
+			public:
+				virtual node_id_t homenode (char* const ptr) {
+					static constexpr std::size_t zero = 0;
+					static const std::size_t prime = (3 * base_distribution<instance>::nodes) / 2;
```

Comment:
This is confusing as it does not return a prime number for most number of nodes. Need to figure out what the limitations are as per our discussions and either document this and prevent execution with the wrong number of nodes, or delay the inclusion of the prime_mapp distributions in this PR.

Reply:
The comment on the PR that the _prime-mapp_ policies work only for a specific number of nodes was misleading.
The bug was fixed some time ago, and the _prime-mapp_ policies work for any number of nodes.
Of course, Jenkins tests will justify this claim.

Reply:
Ok, this is fine as long as all tests and benchmarks pass.